lib/jsdom/browser/xhr/utils.js

```diff
@@ -0,0 +1,235 @@
+"use strict";
+
+var request = require("request");
+var url = require("url");
+var EventEmitter = require("events").EventEmitter;
+var fs = require("fs");
+
+var FormData = require("./formdata");
+var Blob = require("../file/blob");
+
+var utils = {};
+
+function wrapCookieJarForRequest(cookieJar) {
+  var jarWrapper = request.jar();
+  jarWrapper._jar = cookieJar;
+  return jarWrapper;
+}
+
+function getRequestHeader(requestHeaders, header) {
+  var keys = Object.keys(requestHeaders);
+  var n = keys.length;
+  while (n--) {
+    var key = keys[n];
+    if (key.toLowerCase() === header.toLowerCase()) {
+      return requestHeaders[key];
+    }
+  }
+  return null;
+}
+
+utils.getRequestHeader = getRequestHeader;
+
+var base64Regexp = /^(?:[A-Z0-9+\/]{4})*(?:[A-Z0-9+\/]{2}==|[A-Z0-9+\/]{3}=|[A-Z0-9+\/]{4})$/i;
+
+utils.createClient = function createClient(settings, body, callback, callbackError) {
```

Comment:
Can you document what a "client" is, and how it behaves? It's a bit unclear why this is necessary.